devcycle_python_sdk/managers/event_queue_manager.py

```diff
@@ -1,28 +1,111 @@
-import json
+import threading
+import time
 import logging
+import json
 
 from devcycle_python_sdk.dvc_options import DevCycleLocalOptions
 from devcycle_python_sdk.api.local_bucketing import LocalBucketing
+from devcycle_python_sdk.api.event_client import EventAPIClient
+from devcycle_python_sdk.exceptions import (
+    APIClientError,
+    APIClientUnauthorizedError,
+    NotFoundError,
+)
+from devcycle_python_sdk.models.event import FlushPayload, EventType
+
 
 logger = logging.getLogger(__name__)
 
 
-class EventQueueManager:
+class EventQueueManager(threading.Thread):
     def __init__(
         self,
         sdk_key: str,
         options: DevCycleLocalOptions,
         local_bucketing: LocalBucketing,
     ):
-        self.sdk_key = sdk_key
-        self.options = options
-        self.local_bucketing = local_bucketing
+        super().__init__()
+
+        if sdk_key is None or sdk_key == "":
+            raise ValueError("DevCycle is not yet initialized to publish events.")
+
+        self._sdk_key = sdk_key
+        self._options = options
+        self._local_bucketing = local_bucketing
+        self._event_api_client = EventAPIClient(self._sdk_key, self._options)
+        self._flush_lock = threading.Lock()
+
+        # Setup the event queue inside the WASM module
+        event_options_json = json.dumps(self._options.event_queue_options())
+        self._local_bucketing.init_event_queue(event_options_json)
+
+        # Only start event processing thread if event logging is enabled
+        if not (
+            self._options.disable_custom_event_logging
+            and self._options.disable_automatic_event_logging
+        ):
+            self._processing_enabled = True
+            self.daemon = True
+            self.start()
+        else:
+            self._processing_enabled = False
+
+    def _flush_events(self) -> int:
+        if self._flush_lock.locked():
+            return 0
```

Comment:
What's the above check for? It looks like it's designed to drop flushes when a flush is already occurring, but it seems like we only call flush from the main loop.